packages/core/strapi/lib/services/entity-service/types/params/publication-state.ts

```diff
@@ -0,0 +1,30 @@
+import type { Common, Utils } from '@strapi/strapi';
+
+export type Kind = 'preview' | 'live';
+
+export type IsEnabled<TSchemaUID extends Common.UID.Schema> = Utils.Expression.MatchFirst<
+  [
+    [
+      Common.UID.IsContentType<TSchemaUID>,
+      Utils.Expression.IsTrue<Common.Schemas[TSchemaUID]['options']['draftAndPublish']>
+    ],
+    [Common.UID.IsComponent<TSchemaUID>, Utils.Expression.False]
+  ],
+  Utils.Expression.BooleanValue
+>;
+
+export type For<TSchemaUID extends Common.UID.Schema> = IsEnabled<TSchemaUID> extends infer TEnabled
```

Comment:
I'm not sure I see why we need the infer here ? can't we just check if IsEnabled extends true or false directly ?

Reply:
IsEnabled returns true | false if the types are not resolved (not generated usually)
We need a custom case for this situation

Reply:
(and the infer is here because we need to test it against multiple values => avoid too many computation/perfs opti)